### PR TITLE
Refactor: migrate sendbird hooks to ts

### PR DESCRIPTION
**Pull request by HoonBaek**
### Description Of Changes

* Migrate `thunks` and `dux` of Sendbird to TypeScript

[UIKIT-3396](https://sendbird.atlassian.net/browse/UIKIT-3396)

-------------
Let me take out some logics from Sendbird and make custom hooks in separate PR

[UIKIT-3396]: https://sendbird.atlassian.net/browse/UIKIT-3396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/dux/sdk/actionTypes.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { ObjectValues } from "../../../utils/typeHelpers/objectValues";
+
+const actionTypes = {
```
Comment (sravan-s) [inline src/lib/dux/sdk/actionTypes.ts:3]: nit, call this `ACTION_TYPES`

Comment (sravan-s) [reply src/lib/dux/sdk/actionTypes.ts:3]: Lets make that our convention ~

File: src/lib/dux/sdk/reducers.ts [hunk 2]
```diff
@@ -1,13 +1,19 @@
-import * as sdkActions from './actionTypes';
-import initialState from './initialState';
+import SendbirdChat from '@sendbird/chat';
+import sdkActions, { SdkStoreActionTypes } from './actionTypes';
+import initialState, { SdkStoreInitialState } from './initialState';
 
-export default function reducer(state, action) {
+type ActionType = {
```
Comment (sravan-s) [inline src/lib/dux/sdk/reducers.ts:5]: There is an idea to make these type safe, I will share a bit later ~

File: src/lib/dux/sdk/actionTypes.ts (hunk 1 above)
Comment (HoonBaek) [reply src/lib/dux/sdk/actionTypes.ts:3]: Ok ok

Comment (HoonBaek) [reply src/lib/dux/sdk/actionTypes.ts:3]: I think It's good to change all in the different PR

File: src/lib/dux/sdk/thunks.ts
Comment (sravan-s) [inline src/lib/dux/sdk/thunks.ts:1]: Move this into a hook :)

File: src/lib/SendbirdState.tsx [hunk 3]
```diff
@@ -3,8 +3,9 @@ import type { User } from '@sendbird/chat';
 import type { GroupChannel, Member } from '@sendbird/chat/groupChannel';
 import { Dispatch } from 'react';
 import { MarkAsReadSchedulerType } from './hooks/useMarkAsReadScheduler';
+import { ChannelActionTypes } from '../smart-components/Channel/context/dux/actionTypes';
```
Comment (sravan-s) [inline src/lib/SendbirdState.tsx:6]: Lets not import dependencies from modules into parent(SendbirdProvider)

File: src/lib/SendbirdState.tsx [hunk 4]
```diff
@@ -3,8 +3,9 @@ import type { User } from '@sendbird/chat';
...
 
-export type CustomUseReducerDispatcher = Dispatch<{ type: string, payload: any }>;
+export type CustomUseReducerDispatcher = Dispatch<{ type: ChannelActionTypes, payload: any }>;
```
Comment (sravan-s) [inline src/lib/SendbirdState.tsx:8]: This makes all the actionTypes into: `RESET_MESSAGES" | "FETCH_INITIAL_MESSAGES_START" | "FETCH_INITIAL_MESSAGES_SUCCESS" | "FETCH_INITIAL_MESSAGES_FAILURE" | "FETCH_PREV_MESSAGES_SUCCESS" | "FETCH_PREV_MESSAGES_FAILURE" etc...`  becuase this is a patch, just call it string.. We can refactor reducers using the pattern from https://github.com/sendbird/sendbird-uikit-react/pull/486

File: src/lib/dux/sdk/reducers.ts (hunk 2 above)
Comment (sravan-s) [reply src/lib/dux/sdk/reducers.ts:5]: https://github.com/sendbird/sendbird-uikit-react/pull/486

**Comment by AhyoungRyu** [comment]
Oh there're some conflict files. Plz ping me once they're resolved 😄

**Comment by sravan-s** [comment]
@HoonBaek  lets close this, I will help

**Pull request closed without merging**